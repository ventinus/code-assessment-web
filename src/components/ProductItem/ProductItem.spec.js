import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Product, ProductItem } from '..'

let actions = {
  onAddToCartClicked: jest.fn()
}

const setup = product => {
  const component = shallow(
    <ProductItem product={product} {...actions} />
  )

  return {
    product: component.find(Product)
  }
}

let productProps

describe('ProductItem component', () => {
  beforeEach(() => {
    productProps = {
      title: 'Product 1',
      price: 9.99,
      inventory: 6,
    }
  })

  it('should render product', () => {
    const { product } = setup(productProps)
    expect(product.props()).toEqual({ title: 'Product 1', price: 9.99, inventory: 6, onAddToCartClicked: actions.onAddToCartClicked })
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(<ProductItem product={productProps} onAddToCartClicked={actions.onAddToCartClicked} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
