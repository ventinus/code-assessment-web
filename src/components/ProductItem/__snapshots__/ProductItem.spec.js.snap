// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem component should match snapshot 1`] = `
<li
  className="product-item"
>
  <article
    className="product"
  >
    <img
      className="product__img"
    />
    <div
      className="product__body"
    >
      <div
        className="product__body__info"
      >
        <h2
          className="type--a3"
        >
          Product 1
        </h2>
        <p
          className="type--a4"
        >
          $9.99
        </p>
      </div>
      <p
        className="product__body__inventory type--a9 type--grey-6"
      >
        6
         remaining
      </p>
      <button
        className="button type--a8  button--primary product__body__btn"
        disabled=""
        onClick={[Function]}
        type="button"
      >
        Add to cart
      </button>
    </div>
  </article>
</li>
`;
